asmcomp/clambda.ml

```diff
@@ -150,17 +152,21 @@ let rank_structured_constant = function
   | Uconst_float _ -> 0
   | Uconst_int32 _ -> 1
   | Uconst_int64 _ -> 2
-  | Uconst_nativeint _ -> 3
-  | Uconst_block _ -> 4
-  | Uconst_float_array _ -> 5
-  | Uconst_string _ -> 6
-  | Uconst_closure _ -> 7
+  | Uconst_uint32 _ -> 3
+  | Uconst_uint64 _ -> 4
+  | Uconst_nativeint _ -> 5
+  | Uconst_block _ -> 6
+  | Uconst_float_array _ -> 7
+  | Uconst_string _ -> 8
+  | Uconst_closure _ -> 9
```

Comment:
(`rank_structured_constant` is only used to do tag-based comparison of values of different tags, so changing the outputs is fine as long as the relative ordering is preserved. The function is not exported outside the module so there are no compatibility issues.)